doc/docs/master/deploy-manage/deploy/ibmcloud-openshift.md

```diff
@@ -0,0 +1,529 @@
+Pachyderm's default manifests use ports below 1024, so you'll have to modify the manifests to use other port numbers.
+It's usually as easy as adding a "1" in front of the port numbers we use.
+
+#### Pod ports (targetPort)
+
+This is the port exposed by the pod to Kubernetes, which is forwarded to the `port`.
+You should leave the `targetPort` set at `0` so it will match the `port` definition. 
+
+#### External ports (nodePorts)
+
+This is the port accessible from outside of Kubernetes.
+You probably don't need to change `nodePort` values unless your network security requirements or architecture requires you to change to another method of access. 
+Please see the [Kubernetes services documentation](https://kubernetes.io/docs/concepts/services-networking/service/) for details.
+
+## The OCPify script
+
+A bash script that automates many of the substitutions below is available [at this gist](https://gist.github.com/gabrielgrant/86c1a5b590ae3f4b3fd32d7e9d622dc8). 
+You can use it to modify a manifest created using the `--dry-run` flag to `pachctl deploy custom`, as detailed below, and then use this guide to ensure the modifications it makes are relevant to your OpenShift environment.
+It requires certain prerequisites, just as [jq](https://github.com/stedolan/jq) and [sponge, found in moreutils](https://joeyh.name/code/moreutils/).
+
+This script may be useful as a basis for automating redeploys of Pachyderm as needed. 
+
+### Best practices: Infrastructure as code
+
+We highly encourage you to apply the best practices used in developing software to managing the deployment process.
+
+1. Create scripts that automate as much of your processes as possible and keep them under version control.
+1. Keep copies of all artifacts, such as manifests, produced by those scripts and keep those under version control.
+1. Document your practices in the code and outside it.
+
+## Preparing to deploy Pachyderm
+
+Things you'll need
+1. Your storage class
+
+1. Your object store information.
+
+1. Your project in OpenShift.
+
+1. A text editor for editing your deployment manifest.
+## Deploying Pachyderm
+### 1. Setting up PV and object stores
+How you deploy Pachyderm on OpenShift is largely going to depend on where OpenShift is deployed. 
+Below you'll find links to the documentation for each kind of deployment you can do.
+Follow the instructions there for setting up persistent volumes and object storage resources.
+Don't yet deploy your manifest, come back here after you've set up your PV and object store.
+* OpenShift Deployed on [AWS](https://pachyderm.readthedocs.io/en/latest/deployment/amazon_web_services.html) 
+* OpenShift Deployed on [GCP](https://pachyderm.readthedocs.io/en/latest/deployment/google_cloud_platform.html)
+* OpenShift Deployed on [Azure](https://pachyderm.readthedocs.io/en/latest/deployment/azure.html)
+* OpenShift Deployed [on-premise](https://pachyderm.readthedocs.io/en/latest/deployment/on_premises.html)
```

Comment:
Does this portion need to be changed or removed since we are specifically talking about OpenShift in this one?

Reply:
This could be removed as well since we are talking about IBM Cloud specifically.

Reply:
```suggestion
```